lib/ansible/parsing/vault/lib.py

```diff
@@ -0,0 +1,328 @@
+        if not C.DEFAULT_VAULT_ID_MATCH:
+            # Add all of the known vault_ids as candidates for decrypting a vault.
+            vault_id_matchers.extend([_vault_id for _vault_id, _dummy in self.secrets if _vault_id != vault_id])
+
+        matched_secrets = match_secrets(self.secrets, vault_id_matchers)
+
+        # for vault_secret_id in vault_secret_ids:
+        for vault_secret_id, vault_secret in matched_secrets:
+            display.vvvvv(f"Trying to use vault secret=({to_text(vault_secret)}) "
+                          f"id={to_text(vault_secret_id)} to decrypt {to_text(filename)}")
+
+            try:
+                # secret = self.secrets[vault_secret_id]
+                display.vvvv(f"Trying secret {to_text(vault_secret)} for vault_id={to_text(vault_secret_id)}")
+                b_plaintext = this_cipher.decrypt(b_vaulttext, vault_secret)
+                if b_plaintext is not None:
+                    vault_id_used = vault_secret_id
+                    vault_secret_used = vault_secret
+                    file_slug = ''
+                    if filename:
+                        file_slug = f' of "{filename}"'
+                    display.vvvvv(f"Decrypt {to_text(file_slug)} successful with secret={to_text(vault_secret)} "
+                                  f"and vault_id={to_text(vault_secret_id)}")
+                    break
+            except AnsibleVaultFormatError as exc:
+                exc.obj = obj
+                msg = "There was a vault format error"
+                if filename:
+                    msg += f' in {(to_text(filename))}'
+                msg += f': {to_text(exc)}'
+                display.warning(msg, formatted=True)
+                raise
+            except AnsibleError as e:
+                display.vvvv(f'Tried to use the vault secret ({to_text(vault_secret_id)}) to decrypt '
+                             f'({to_text(filename)}) but it failed. Error: {e}')
+                continue
+        else:
+            msg = "Decryption failed (no vault secrets were found that could decrypt)"
+            if filename:
+                msg += f" on {to_native(filename)}"
+            raise AnsibleVaultError(msg)
+
+        if b_plaintext is None:
+            msg = "Decryption failed"
+            if filename:
+                msg += f" on {to_native(filename)}"
+            raise AnsibleError(msg)
+
+        return b_plaintext, vault_id_used, vault_secret_used
+
+
+def is_encrypted(data):
+    """ Test if this is vault encrypted data blob
+
+    :arg data: a byte or text string to test whether it is recognized as vault
+        encrypted data
+    :returns: True if it is recognized.  Otherwise, False.
+    """
+    try:
+        # Make sure we have a byte string and that it only contains ascii
+        # bytes.
+        b_data = to_bytes(to_text(data, encoding='ascii', errors='strict', nonstring='strict'), encoding='ascii',
+                          errors='strict')
```

Comment:
```suggestion
        b_data = to_bytes(to_text(data, encoding='ascii', errors='strict', nonstring='strict'), encoding='ascii', errors='strict')
```

Reply:
this would cause tests to fail, as the line is too long

Reply:
It shouldn't. This was the original line, which is < 160 characters.